training/src/main/scala/com/airbnb/aerosolve/training/NDTree.scala

```diff
@@ -61,16 +64,33 @@ object NDTree {
     if ((indices.length > 0) && (dimensions > 0) && isConstantDimension) {
       val node = new NDTreeNode()
       nodes.append(node)
-      buildTreeRecursive(options, points, indices, nodes, node, 1, splitType)
+      buildTreeRecursive(options, points, -1, indices, nodes, node, 1, splitType)
     }
 
     val tree = new NDTree(nodes.toArray)
     tree
   }
 
+  def getNextAxis(lastAxis: Int,
+                  deltas: Array[Double],
+                  node: NDTreeNode,
+                  minLeafWidthPercentage: Double):Int = {
+    for(i <- 1 to deltas.length) {
+      val axis = (lastAxis + i) % deltas.length
+      val rootWidth = node.getMax.get(axis) - node.getMin.get(axis)
+      val width = deltas(axis)
+      if (rootWidth > 0 && width > 0 &&
+          minLeafWidthPercentage * rootWidth <= width) {
+        return axis
```

Comment:
@saurfang Thanks for review! updated and comments added. For this for loop case, if I use find,     List.range(1, deltas.length).find()
the logic in getNextAxis seems hard to bind in, since find just return the i, then I need to construct axis from i, but inside find, I need to construct it too, am I missing anything?

and Yes, I agree, it needs a  high level description, I will do it right after model finish, I am still research what's the right way to construct the tree.... in a debug mode.